common/trinary/trit_long.c

```diff
@@ -56,7 +59,7 @@ size_t long_to_trits(int64_t const value, trit_t *const trits) {
 
   negative = value < 0;
   size = min_trits(value);
-  int64_t v_abs = negative ? -value : value;
+  uint64_t v_abs = (uint64_t)value;
```

Comment:
You should use `llabs` here. While not ideal, casting a negative value to an int64 is not equal to its absolute value